### ISSUE TITLE
Blazor server Safari fullscreen issue

### ISSUE THREAD

**Issue by pinkplus** [dotnet/aspnetcore#39726]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I'm making a Blazor server app. I'm adding an event handler to enable fullscreen mode, by creating a JS function and calling it via JS interop. It works on Chrome, but it doesn't seem to work on Safari desktop.

I've made a minimal repro project. The JS functions are defined [here](https://github.com/pinkplus/SafariFullscreen/blob/master/Pages/Index.razor.js), and the component is defined [here](https://github.com/pinkplus/SafariFullscreen/blob/master/Pages/Index.razor).

### Expected Behavior

Fullscreen should be able to be enabled in Blazor server event handler.

### Steps To Reproduce

This is the repro repo: https://github.com/pinkplus/SafariFullscreen.

1. Clone the repo, and run `dotnet run`.
2. Open http://localhost:5071 in Safari. You should be able to see two buttons: "Enter fullscreen javascript" and "Enter fullscreen blazor".
3. Click the "Enter fullscreen javascript" button. You should be in fullscreen mode now. Press ESC to escape fullscreen mode.
4. Click the "Enter fullscreen blazor" button. Fullscreen mode should be enabled, but it's not.

### .NET Version

6.0.101

### Anything else?

I'm using the latest Safari (15.2) on the latest macOS (12.1).

I have a feeling that this may be related to Safari's restriction. Safari may require a user gesture event so that fullscreen can be enabled. With Blazor server, requestFullscreen is probably called in a websocket event handler, so it doesn't work.

**Comment by pinkplus** [comment]
@javiercn Yes, it works on a static HTML page. In the repro repo I've included a button using pure javascript as well. You can see it works well with pure javascript.

**Comment by KristofferStrube** [comment]
Hey @pinkplus,
In your project you don't await the asynchronous call here: https://github.com/pinkplus/SafariFullscreen/blob/master/Pages/Index.razor#L25

I don't know if that should make a difference but I don't have an Apple device so I can't test it.

**Comment by pinkplus** [comment]
Hi @KristofferStrube,

It doesn't seem to make a difference. I've updated the project to await the call though.

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us.
This is because the `requestFullscreen` API requires user gesture, and all Blazor Server interactions are asynchronous. This is well described in [this](https://javascript.plainenglish.io/user-gesture-restricted-web-apis-d794454453f7) article.
As a workaround we recommend handling the user gesture event directly in JavaScript.

Here what you can try to do:
```razor
onclick="this.requestFullScreeen()"
```

Note: that the `onclick` is not `@onclick` in this case.
